BlogTemplate.Tests/Models/BlogDataStoreTests.cs

```diff
@@ -234,6 +234,82 @@ public void GetAllPosts_ReturnsList()
             Assert.NotEmpty(posts);
         }
 
+        [Fact]
+        public void UpdatePost_ChangePost_UpdatesXMLFile()
+        {
+            BlogDataStore testDataStore = new BlogDataStore();
+
+            Post oldPost = new Post
+            {
+                Slug = "Old-Title",
+                Title = "Old Title",
+                Body = "Old body",
+                IsPublic = true,
+                Excerpt = "Old excerpt"
+            };
+
+            Post newPost = new Post
+            {
+                Slug = "New-Title",
+                Title = "New Title",
+                Body = "New body",
+                IsPublic = true,
+                Excerpt = "New excerpt"
+            };
+
+            testDataStore.SavePost(oldPost);
+            testDataStore.UpdatePost(newPost, oldPost);
+
+            Assert.True(File.Exists($"BlogFiles//New-Title.xml"));
+            Post result = testDataStore.CollectPostInfo($"BlogFiles//New-Title.xml", "New-Title");
+            Assert.Equal(result.Slug, "New-Title");
+            Assert.Equal(result.Title, "New Title");
+            Assert.Equal(result.Body, "New body");
+            Assert.True(result.IsPublic);
+            Assert.Equal(result.Excerpt, "New excerpt");
+        }
+
+        [Fact]
+        public void UpdatePost_ChangePost_DoesNotRemoveComments()
+        {
+            BlogDataStore testDataStore = new BlogDataStore();
+
+            Post oldPost = new Post
+            {
+                Slug = "Old-Title",
+                Title = "Old Title",
+                Body = "Old body",
+                IsPublic = true,
+                Excerpt = "Old excerpt"
+            };
+            Comment comment = new Comment
+            {
+                AuthorName = "Test name",
+                AuthorEmail = "Test email",
+                Body = "test body",
+                PubDate = DateTime.Now,
+                IsPublic = true
+            };
+            Post newPost = new Post
+            {
+                Slug = "New-Title",
+                Title = "New Title",
+                Body = "New body",
+                IsPublic = true,
+                Excerpt = "New excerpt"
+            };
+
+            testDataStore.SavePost(oldPost);
+            testDataStore.SaveComment(comment, oldPost);
```

Comment:
I talked to Lucas about removing this.  testDataStore.SavePost() will save all of the comments as well, so you could do oldPost.Comments.Add(comment) before calling SavePost.

Or, you could do it inline
```
Post oldPost = new Post
{
    Slug = "..."
    Comments = new List<Comment>
    {
        new Comment
        {
            AuthorEmail = "johnsmith@example.com",
            AuthorName = "John Smith",
            Body = "Sample Comment",
        }
    }
}
```